### PR TITLE
chore: Reuses project in tests - project

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Reuses project in tests - project.

Moves data source tests with resource tests.

Link to any related issue(s): CLOUDP-239550

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/project/resource_project_test.go [hunk 1]
```diff
@@ -491,22 +493,50 @@ func TestResourceProjectDependentsDeletingRefreshFunc(t *testing.T) {
...
+		orgID          = os.Getenv("MONGODB_ATLAS_ORG_ID")
+		projectName    = acc.RandomProjectName()
+		projectOwnerID = os.Getenv("MONGODB_ATLAS_PROJECT_OWNER_ID")
 	)
+	commonChecks := map[string]string{
+		"name":          projectName,
+		"org_id":        orgID,
+		"cluster_count": "0",
+		"teams.#":       "2",
+	}
```
Comment (AgustinBettati) [inline internal/service/project/resource_project_test.go:514]: could we add a check for `limits` attribute as well? seeing that it was tested in the data source tests

**Review by AgustinBettati — APPROVED** [review]
LGTM, small comment

**Review by oarbusi — APPROVED** [review]

File: internal/service/project/resource_project_test.go (hunk 1 above)
Comment (lantoli) [reply internal/service/project/resource_project_test.go:514]: sure, i'll do in the limit test

File: internal/service/project/resource_project_test.go [hunk 2]
```diff
@@ -823,19 +804,35 @@ func TestAccProjectRSProject_importBasic(t *testing.T) {
 	})
 }
 
-func TestAccProjectRSProject_withUpdatedLimits(t *testing.T) {
+func TestAccProject_withUpdatedLimits(t *testing.T) {
 	var (
 		orgID       = os.Getenv("MONGODB_ATLAS_ORG_ID")
 		projectName = acc.RandomProjectName()
+		limitChecks = []map[string]string{
```
Comment (lantoli) [inline internal/service/project/resource_project_test.go:811]: @AgustinBettati ds limit tests

**Review by maastha — APPROVED** [review]

**Review by AgustinBettati — APPROVED** [review]

**Pull request merged**